fendermint/vm/interpreter/src/fvm/checkpoint.rs

```diff
@@ -212,3 +241,164 @@ where
 
     Ok(PowerTable(power_table))
 }
+
+/// Convert the collaterals and metadata in the membership to the public key and power expected by the system.
+fn membership_to_power_table(m: &Membership, power_scale: PowerScale) -> PowerTable {
+    let mut pt = Vec::new();
+
+    for v in m.validators.iter() {
+        // Ignoring any metadata that isn't a public key.
+        if let Ok(pk) = PublicKey::parse_slice(&v.metadata, None) {
+            let c = u256_to_tokens(v.weight);
+            pt.push(Validator {
+                public_key: ValidatorKey(pk),
+                power: Collateral(c).into_power(power_scale),
+            })
+        }
+    }
+
+    PowerTable(pt)
+}
+
+fn u256_to_tokens(value: U256) -> TokenAmount {
```

Comment:
Nice! I'm sure I have done this somewhere too... heah, [here](https://github.com/consensus-shipyard/fendermint/blob/995e193f3b97e816b2b9e020760312a4a5a25bab/fendermint/eth/api/src/conv/from_tm.rs#L45). I'm sure it's the easiest to debug, while this may be faster, who knows. 

I did it this way because it was symmetrical with [tokens_to_u256](https://github.com/consensus-shipyard/fendermint/blob/main/fendermint/vm/actor_interface/src/ipc.rs#L234), but the other way I also have a [safer](https://github.com/consensus-shipyard/fendermint/blob/995e193f3b97e816b2b9e020760312a4a5a25bab/fendermint/vm/message/src/conv/from_fvm.rs#L29) conversion that checks maximum boundaries. 

It would be nice if these were in a common place, just haven't found a good one yet.

Reply:
Yep, I think you approach is better. Maybe we can have these either in the actors bindings or in the `ipc-agent/sdk`. That being said, for now I think is fine where it is.